src/tiled/layeroffsettool.cpp

```diff
@@ -136,7 +136,8 @@ void LayerOffsetTool::startDrag(const QPointF &pos)
     if (!mapDocument())
         return;
 
-    if (Layer *layer = mapDocument()->currentLayer()) {
+    Layer *layer = mapDocument()->currentLayer();
+    if (layer && layer->isUnlocked()) {
```

Comment:
I noticed that this change is not enough, since after calling `startDrag`, the `mouseMoved` function continues to change the layer position even if `mDragging` is not set to `true`. So I would suggest to remove the `return` in the `else` in that function and instead check `mDragging` in addition to `currentLayer` before moving the layer.